test/schemaValidation.test.ts

```diff
@@ -1967,4 +1967,54 @@ obj:
       expect(telemetry.messages).to.be.empty;
     });
   });
+  it('Nested AnyOf const should correctly evaluate and merge problems', async () => {
+    // note that 'missing form property' is necessary to trigger the bug (there has to be some problem in both subSchemas)
+    // order of the object in `anyOf` is also important
+    const schema: JSONSchema = {
+      type: 'object',
+      properties: {
+        options: {
+          anyOf: [
+            {
+              type: 'object',
+              properties: {
+                form: {
+                  type: 'string',
+                },
+                provider: {
+                  type: 'string',
+                  const: 'test1',
+                },
+              },
+              required: ['form', 'provider'],
+            },
+            {
+              type: 'object',
+              properties: {
+                form: {
+                  type: 'string',
+                },
+                provider: {
+                  anyOf: [
+                    {
+                      type: 'string',
+                      const: 'testX',
+                    },
+                  ],
+                },
+              },
+              required: ['form', 'provider'],
+            },
+          ],
+        },
+      },
+    };
+    languageService.addSchema(SCHEMA_ID, schema);
```

Comment:
Can you rebase this commit to head and use `schemaProvider.addSchema` it is the new mechanism for feeding schemas for tests and since we have removed the earlier hack in the parser this will probably fail.

Reply:
thanks, @gorkem done,

I also pushed a small change for const matching.
there was a small inconsistency between `enum` and `const` match.
Enum allows having a partial match, but const had to be exactly the same.

It's a little bit related to this PR, so sorry for putting it together...